src/main/java/de/natrox/common/container/Triple.java

```diff
@@ -92,4 +92,14 @@ public Triple<A, B, C> setThird(C third) {
         this.third = third;
         return this;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (!obj.getClass().equals(this.getClass()))
+            return false;
+        Triple<?, ?, ?> that = (Triple<?, ?, ?>) obj;
+        return this.first() == that.first() && this.second() == that.second() && this.third() == that.third();
```

Comment:
Attention == is not the same as #equals. Since you are in an equals method, I think the values should rather be compared to #equals.